### PR TITLE
[build] Remove unemployed 'skip-{build,test}-*' flags from build-script-impl

### PR DESCRIPTION
**Pull request by finagolfin**
These flags have done nothing for awhile, as 26 of 31 have already been moved to `build-script`. Only the flags related to haiku, maccatalyst, and external-benchmarks are unrecognized by either build script after this removal.

I just checked to see if `--skip-build-linux` would disable building all products for that platform, as [I've been doing manually when cross-compiling the toolchain for Android](https://github.com/buttaface/termux-packages/commit/ac665480a93f0d9680dc9dc32f650e60017bce91#diff-7112cf52b0d0e20b7676ce6f86e56cfdR72), and was surprised to find that the flag did nothing in `build-script-impl`. I investigated further and found that these 30 flags do nothing too, as most of them have been moved to `build-script` and their functionality removed from the bash script, eg #26495.

The haiku-related functionality has been swept aside over time, while [maccatalyst never did anything](https://github.com/apple/swift/pull/29017/files#diff-65b44eb6cb88af2161d8e1176231aad0R150), maybe because the actual implementation hasn't been open-sourced? I noticed that [another flag from that Catalyst pull does nothing](https://github.com/apple/swift/pull/29017/files#diff-23cfac4b9c8ced6111824fa8012556e2R604), but didn't remove it here.

@gottesmm, all the implementation of `--build-external-benchmarks` has been moved to `build-script` and the Python scripts. You could add back a `--skip-build-external benchmarks` there that actually works, but since the default is to skip it, probably not needed.

@Rostepher, review requested.

### CODE REVIEW

**Comment by Rostepher** [comment]
@edymtt can probably review this change. I'm no longer working on the Swift project.

**Comment by buttaface** [comment]
@bitjammer, looks like you added a lot of these flags, review requested.

**Comment by bitjammer** [comment]
I haven't edited this file since it was reorganized some years ago. I recommend asking @shahmishal if he or someone else can review on this one. Thanks!

**Comment by shahmishal** [comment]
@swift-ci smoke test

**Comment by shahmishal** [comment]
@swift-ci build toolchain

**Review by compnerd — APPROVED** [review]

**Comment by compnerd** [comment]
@swift-ci please smoke test Linux platform

**Comment by gottesmm** [comment]
@buttaface You can remove all of the cmake build benchmark external.

File: utils/build-script [hunk 1]
```diff
@@ -560,9 +560,6 @@ class BuildScriptInvocation(object):
             impl_args += ["--skip-build"]
         if not args.build_benchmarks:
             impl_args += ["--skip-build-benchmarks"]
-        # Currently we do not build external benchmarks by default.
-        if args.build_external_benchmarks:
```
Comment (gottesmm) [inline utils/build-script:564]: This can go.

File: utils/build-script [hunk 2]
```diff
@@ -596,26 +593,6 @@ class BuildScriptInvocation(object):
         if args.build_swift_static_sdk_overlay:
             impl_args += ["--build-swift-static-sdk-overlay"]
 
-        if not args.build_linux:
-            impl_args += ["--skip-build-linux"]
-        if not args.build_freebsd:
-            impl_args += ["--skip-build-freebsd"]
-        if not args.build_cygwin:
-            impl_args += ["--skip-build-cygwin"]
-        if not args.build_osx:
```
Comment (gottesmm) [inline utils/build-script:605]: Does this really not do anything? I am pretty sure I have used these in the past bit.

**Review by gottesmm — COMMENTED** [review]
A few questions.

File: utils/build-script (hunk 2 above)
Comment (buttaface) [reply utils/build-script:605]: If invoked originally from the Python `build-script`, it hasn't done anything in this bash script in many years, as it's handled in Python, #2880. As a result, the actual implementation of this flag in this bash script was removed last year in the aforementioned pull #26495.

**Comment by buttaface** [comment]
>You can remove all of the cmake build benchmark external.

All bash flags in this pull were removed because I could find no trace of them being used in the bash script `build-script-impl`, ie none are used anywhere or passed to any of the CMake invocations in the script. All usage of these flags has long since been removed, I'm just taking the last step of removing the unused flags.

**Comment by buttaface** [comment]
Ping, ready to go?

**Comment by buttaface** [comment]
Ping @shahmishal, ready to merge, though you may need to rerun the CI with the latest commit from the main branch first.

**Comment by shahmishal** [comment]
@swift-ci test and merge

**Pull request merged**